examples/with-docker/Dockerfile

```diff
@@ -3,12 +3,15 @@ FROM node:16-alpine AS deps
 # Check https://github.com/nodejs/docker-node/tree/b4117f9333da4138b03a546ec926ef50a31506c3#nodealpine to understand why libc6-compat might be needed.
 RUN apk add --no-cache libc6-compat
 WORKDIR /app
-COPY package.json yarn.lock ./
-RUN yarn install --frozen-lockfile
 
-# If using npm with a `package-lock.json` comment out above and use below instead
-# COPY package.json package-lock.json ./ 
-# RUN npm ci
+# Install dependencies based on the preferred package manager
+COPY package.json yarn.lock* package-lock.json* pnpm-lock.yaml* ./
+RUN \
+  [ -f yarn.lock ] && yarn --frozen-lockfile --prod || \
+  [ -f package-lock.json ] && npm ci || \
+  [ -f pnpm-lock.yaml ] && yarn global add pnpm && pnpm fetch --prod && pnpm i -r --offline --prod || \
```

Comment:
There is `wget -qO- https://get.pnpm.io/install.sh | sh -` also, but I got some errors while trying that locally, so I stuck with `yarn global add pnpm`. Is this a bad idea? :thinking:

Reply:
Why not `npm i pnpm -g`🤔?
https://pnpm.io/installation#using-npm

Reply:
yarn might be slight faster, I think. haven't tested it though